internal/reconciliations/oathkeeper/deployment.go

```diff
@@ -133,3 +128,13 @@ func getReplicasForDeployment(ctx context.Context, k8sClient client.Client) (str
 
 	return strconv.Itoa(int(*dep.Spec.Replicas)), nil
 }
+
+func hasMinimumReplicasAvailable(dep appsv1.Deployment) bool {
```

Comment:
With the new approach we want wait for the restart to finish. As so, for example a CrashLoopBackoff of new pods during a rollout restart will not be seen as an error (as the deployment is in ConditionTrue during the whole restart/upgrade). Only the ReplicaSet condition is in progressing state. Are we fine with that?

Reply:
I think with waiting for the deployment to be available we are already crossing the line of responsibility of the operator a little bit. 
In my opinion the operator has the responsibility to reconcile the cluster to an expected state. In our case one part of it is, that Oathkeeper resources are applied. Additionally we wait for the deployment to be Ready/Available. This validation of availability can already be seen as part of the monitoring. And monitoring is not a good use case for the reconciliation, because it succeeded, the next default reconciliation will happen in approx. 10h and until than no "monitoring" will happen.
If we not only monitor the availability of the deployment, but also the status of the ReplicaSet, we go even deeper into misusing reconciliation for monitoring.

Reply:
Yes, in that terms it makes sense.